### DISCUSSION TITLE
Unable to compile MOOSE due to PETSc error in Linux

### DISCUSSION THREAD

**Discussion by sb50nemo** [idaholab/moose#17228] [Q&A Getting Started] [answered]
I am trying to install MOOSE on my linux machine by following the procedure in [MOOSE Framework - Getting started.](https://mooseframework.inl.gov/getting_started/installation/conda.html). However, I am unable to compile MOOSE because of the following error:

`/home/user/miniconda3/envs/moose/include/petscsys.h:211:6: error: #error "PETSc was configured with MPICH but now appears to be compiling using a non-MPICH mpi.h"`
`  211 | #    error "PETSc was configured with MPICH but now appears to be compiling using a non-MPICH mpi.h"
      |      ^~~~~
`

How can I deal with this error? Any suggestions would be much appreciated. Thank you.

**Comment by milljm** [comment, marked as answer]
There appears to be multiple MPI wrappers available within your PATH, and `make` is choosing the wrong one.

Can you tell us more about the machine you are operating on? Linux? What Distro/Flavor? Also, a report of what your environment looks like would be very helpful (these sorts of issues are almost always environment related):
```bash
env
```

Do you use `modules`? Any loaded (env above would tell us)?
```bash
module list
```

What does your PATH look like (env above would tell us):
```bash
echo $PATH
```

Cheers!

**Reply by sb50nemo** [reply]
Thanks Jason!

I figured out that I needed to install mpich module, and then I was able to install MOOSE.

Thanks a lot for your help!

**Reply by richman2024** [reply]
Hi, I found multiple MPI wrappers are available in my PATH, how could I choose the correct one.

Thank you for your help!

**Reply by GiudGiud** [reply]
Please post in a new discussions thread. 
We ll need to know your installation method and platform

**Reply by richman2024** [reply]
I post a new discusion as #25047
